command/ca/provisioner/webhook/webhook.go

```diff
@@ -0,0 +1,126 @@
+
+## EXAMPLES
+
+Add a new webhook to a provisioner:
+'''
+step ca provisioner webhook add my_webhook --provisioner my_provisioner --url https://example.com
+'''
+
+Change a webhook's url:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --url https://example.com
+'''
+
+Remove a webhook:
+'''
+step ca provisioner webhook remove my_webhook --provisioner my_provisioner
+'''
+		`,
+	}
+}
+
+var (
+	provisionerFlag = cli.StringFlag{
+		Name:     "provisioner",
+		Usage:    `The name of the provisioner the webhook is attached to.`,
+		Required: true,
+	}
+	urlFlag = cli.StringFlag{
+		Name:  "url",
+		Usage: `The url of the webhook server.`,
+	}
+	kindFlag = cli.StringFlag{
+		Name:  "kind",
+		Usage: `The kind of webhook. Default is ENRICHING.`,
+	}
+	bearerTokenFileFlag = cli.StringFlag{
+		Name:  "bearer-token-file",
+		Usage: `The token to be set in the Authorization header of the request to the webhook server.`,
+	}
+	basicAuthUsernameFlag = cli.StringFlag{
+		Name:  "basic-auth-username",
+		Usage: `The username portion of the Authorization header of the request to the webhook server when using basic authentication.`,
+	}
+	basicAuthPasswordFileFlag = cli.StringFlag{
+		Name:  "basic-auth-password-file",
+		Usage: `The password porition of the Authorization header of the request to the webhook server when using basic authentication.`,
+	}
+	disableTLSClientAuthFlag = cli.BoolFlag{
+		Name:  "disable-tls-client-auth",
+		Usage: `The CA will not send a client certificate when requested by the webhook server.`,
+	}
+	certTypeFlag = cli.StringFlag{
+		Name:  "cert-type",
+		Usage: `Whether to call this webhook when signing X509 certificates, SSH certificates, or ALL certificates. Default is ALL.`,
+	}
+)
+
+type crudClient interface {
+	GetProvisioner(...ca.ProvisionerOption) (*linkedca.Provisioner, error)
+	CreateProvisionerWebhook(provisionerName string, wh *linkedca.Webhook) (*linkedca.Webhook, error)
+	UpdateProvisionerWebhook(provisionerName string, wh *linkedca.Webhook) (*linkedca.Webhook, error)
+	DeleteProvisionerWebhook(provisionerName string, webhookName string) error
+}
+
+func newCRUDClient(cliCtx *cli.Context, cfgFile string) (crudClient, error) {
+	// os.Stat("") probably returns os.ErrNotExist, but this behavior is
+	// undocumented so we'll handle this case separately.
+	if cfgFile == "" {
+		return cautils.NewAdminClient(cliCtx)
+	}
+
+	_, err := os.Stat(cfgFile)
+	switch {
+	case errors.Is(err, os.ErrNotExist):
+		return cautils.NewAdminClient(cliCtx)
+	case err == nil:
+		ui.PrintSelected("CA Configuration", cfgFile)
+		cfg, err := config.LoadConfiguration(cfgFile)
+		if err != nil {
+			return nil, fmt.Errorf("error loading configuration: %w", err)
+		}
+		if cfg.AuthorityConfig.EnableAdmin {
+			if len(cfg.AuthorityConfig.Provisioners) > 0 {
```

Comment:
Is there a reason to validate this condition here? Feels like something that we should handle globally in config.LoadConfiguration() or not at all (in case for some reason we decide to loosen the requirement).

Reply:
97eef2d
Removed from here. Should I remove from `command/ca/provisioner/provisioner.go` also?